### PR TITLE
Relaxing function signatures to make them work with ForwardDiff

### PR DESCRIPTION
**Pull request by jbrea**
These are the changes I made locally, to use ForwardDiff in the BayesianOptimization package.

One thing I was wondering was whether you want to keep the MatF64 and the VecF64 constants. I could also replace all instances with AbstractMatrix and AbstractVector.

### CODE REVIEW

**Comment by chris-nemeth** [comment]
I don't think it's necessary to keep `MatF64` and `VecF64`. @maximerischard - what you think as I believe you introduced them?

**Comment by maximerischard** [comment]
I’m also in favour of eliminating them. It’s convenient but it makes it harder for others to read the source code.

**Comment by jbrea** [comment]
OK, I think this is ready to be merged.

File: src/GP.jl [hunk 1]
```diff
@@ -9,13 +9,13 @@ Return posterior mean and variance of the Gaussian Process `gp` at specfic point
...
 """
-function predict_f(gp::GPBase, x::MatF64; full_cov::Bool=false)
+function predict_f(gp::GPBase, x::AbstractArray{T, 2}; full_cov::Bool=false) where T
     size(x,1) == gp.dim || throw(ArgumentError("Gaussian Process object and input observations do not have consistent dimensions"))
     if full_cov
         return _predict(gp, x)
     else
         ## Calculate prediction for each point independently
-            μ = Array{Float64}(undef, size(x,2))
+            μ = Array{T}(undef, size(x,2))
```
[the remark this answers is not in the document]
Comment (jbrea) [reply src/GP.jl]: Done.

**Comment by maximerischard** [comment]
This is great, thank you. I merged the PR manually after a bit of git surgery.

**Pull request closed without merging**